controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```
@@ -944,19 +998,58 @@ func getPublishRequestActID(vmPub *vmopv1.VirtualMachinePublishRequest) string {
 
 // parseItemIDFromTaskResult returns the content library item UUID from the task result.
 func parseItemIDFromTaskResult(result vimtypes.AnyType) (string, error) {
-	clItem, ok := result.(vimtypes.ManagedObjectReference)
+	item, ok := result.(vimtypes.ManagedObjectReference)
 	if !ok {
 		return "", fmt.Errorf("failed to cast task result to ManagedObjectReference")
 	}
 
-	if clItem.Type != "ContentLibraryItem" {
-		return "", fmt.Errorf("expect task result type to be ContentLibraryItem, get %s instead", clItem.Type)
+	switch item.Type {
+	case "ContentLibraryItem":
+		if !strings.HasPrefix(item.Value, clItemPrefix) {
+			return "", fmt.Errorf("content library item UUID is invalid, value: %s", item.Value)
+		}
+		return item.Value[len(clItemPrefix):], nil
+	case "VirtualMachine":
+		return item.Value, nil
+	default:
+		return "", fmt.Errorf("expect task result type to be ContentLibraryItem or VirtualMachine, got %s instead", item.Type)
+	}
+}
+
+func (r *Reconciler) getTargetLibrary(ctx *pkgctx.VirtualMachinePublishRequestContext) error {
+	vmPubReq := ctx.VMPublishRequest
+	targetLocationName := vmPubReq.Spec.Target.Location.Name
+	objKey := client.ObjectKey{Name: targetLocationName, Namespace: vmPubReq.Namespace}
+
+	contentLibrary := &imgregv1a1.ContentLibrary{}
+	if err := r.Get(ctx, objKey, contentLibrary); err != nil {
+		ctx.Logger.Error(err, "failed to get ContentLibrary", "cl", objKey)
+		if apierrors.IsNotFound(err) {
+			conditions.MarkError(vmPubReq,
+				vmopv1.VirtualMachinePublishRequestConditionTargetValid,
+				vmopv1.TargetContentLibraryNotExistReason,
+				err)
+		}
```

Should we return the error here? If so, no need to log the error with wrapping.